src/Functions/reinterpretAs.cpp

```diff
@@ -261,8 +262,10 @@ class FunctionReinterpret : public IFunction
                             memcpy(static_cast<void*>(&to[i]), static_cast<const void*>(&from[i]), copy_size);
                         else
                         {
-                            size_t offset_to = sizeof(To) > sizeof(From) ? sizeof(To) - sizeof(From) : 0;
-                            memcpy(reinterpret_cast<char*>(&to[i]) + offset_to, static_cast<const void*>(&from[i]), copy_size);
+                            // Handle the cases of both 128-bit representation to 256-bit and 128-bit to 64-bit or lower.
+                            const size_t offset_from = sizeof(From) > sizeof(To) ? sizeof(From) - sizeof(To) : 0;
```

Comment:
Is this change here intended?

Reply:
Yeah, this looks a little funny but is needed. I came across an interesting interaction between what `memcpy` does on a Big Endian platform and moving between a datatype of higher number of bits to lower. 

I have a line in `tests/queries/0_stateless/02862_uuid_reinterpret_as_numeric.sql` that captures this

```
...
SELECT reinterpretAsUUID(reinterpretAsUInt128(reinterpretAsUInt32(reinterpretAsUInt256(x)))) FROM t_uuid;
...
```

L267, which was there before handles the case of moving from a lower to higher bit number. Ex:

```
[ aaaa bbbb ] (64)
[ 0000 0000 aaaa bbbb ] (128)
```

Without L266 and using `memcpy`, the most significant portion of the number is copied.

```
[ aaaa bbbb cccc dddd ] (128)
[ aaaa bbbb ] (64)
```

The addition of the from offset allows copying the correct portion, replicating the truncating behavior as expected.

```
[ aaaa bbbb cccc dddd ] (128)
[ cccc dddd ] (64)
```

An alternative would have been to do a reverse memory copy, then flip the bits to get the same effect.